pwiz_tools/Skyline/Model/Results/SpectrumFilterPair.cs

```diff
@@ -237,6 +237,18 @@ private ExtractedSpectrum FilterSpectrumList(MsDataSpectrum[] spectra,
                 if (imRangeHelper.IsBeyondRange(spectrum))
                     break;
 
+                if (ReferenceEquals(productFilters, Ms2ProductFilters) &&
+                    spectrum.Metadata != null &&
+                    spectrum.Metadata.ScanWindowLowerLimit.HasValue &&
+                    spectrum.Metadata.ScanWindowUpperLimit.HasValue &&
+                    productFilters.All(f =>

```

Comment:
"productFilters" is always sorted by TargetMz.
The code that ensures they are sorted is in "AddFilterValues".
Therefore, instead of doing "productFilters.All" you can just compare the first product filter to ScanWindowUpperLimit and the last product filter to ScanWindowLowerLimit.